app/services/support_interface/unexplained_breaks_in_work_history_export.rb

```diff
@@ -0,0 +1,108 @@
+module SupportInterface
+  class UnexplainedBreaksInWorkHistoryExport
+    class UnexplainedBreak
+      def initialize(month_range:)
+        @month_range = month_range
+      end
+
+      def start_date
+        @month_range.first.prev_month
+      end
+
+      def end_date
+        @month_range.last.next_month
+      end
+
+      def length
+        ((end_date.year * 12) + end_date.month) - ((start_date.year * 12) + start_date.month)
+      end
+    end
+
+    def data_for_export
+      data_for_export = Candidate.all.map do |candidate|
```

Comment:
I'm aware that this loads all of the candidates in to memory and will be quite slow in production. If anyone is able to suggest an improvement here I'd be interested to know

Reply:
These exports run as a Sidekiq job so hopefully that will alleviate any major memory issues. The only other suggestion I can think of is whether you could use `select` to bring back only the data you require in each application form (e.g. https://github.com/DFE-Digital/apply-for-teacher-training/commit/f8b45fb6dc802ffa273649ab5646e92ca3c39e39#diff-f3cf4595d7d06f89258aa90c9f8553e0a18288e0b7d303469a05efd582e1ff60R31)

Reply:
Perhaps use of [includes](https://apidock.com/rails/v6.0.0/ActiveRecord/QueryMethods/includes) might be an idea also - Candidate.includes(:application_forms)

Reply:
I'm struggling to write a Rails query to get the most recent application for each candidate before I start. 

@TobyRet has helped me to get to  `ApplicationForm.order(:submitted_at).group_by{|a| a.candidate_id}.limit(1)`, but this still isn't quite right as it only returns a single ApplicationForm rather than one for each Candidate.

I've added selects and includes

```
data_for_export = Candidate.includes(:application_forms).all.map do |candidate|
        application_form = candidate.application_forms
                               .select(:submitted_at, :date_of_birth, :candidate_id, :id)
                               .includes(:application_qualifications, :application_work_experiences, :application_work_history_breaks)
                               .order(:submitted_at).last
```
It doesn't make a difference at this scale, but it might once we run it on the production data